packages/text-field/Input.tsx

```diff
@@ -144,6 +144,13 @@ export default class Input<
       setDisabled,
     } = this.props;
 
+    if (
```

Comment:
This condition looks complicated :) 

Do you need to check the previous state too?

Reply:
You need to check previous state because you only want this to trigger when both are set. This only occurs when one was changed from unset --> set.

Reply:
And yeah...it is complicated, but it is a common react pattern to check previous state or props.